lib/client.js

```diff
@@ -86,17 +86,69 @@ async function get(config) {
     );
   }
 
-  let jwks;
-  if (config.clientAssertionSigningKey) {
-    const jwk = JWK.asKey(config.clientAssertionSigningKey).toJWK(true);
-    jwks = { keys: [jwk] };
+  const configTokenAuthMethod = config.clientAuthMethod;
+  const issuerTokenAuthMethods = Array.isArray(
+    issuer.token_endpoint_auth_methods_supported
+  )
+    ? issuer.token_endpoint_auth_methods_supported
+    : [];
+  if (!issuerTokenAuthMethods.includes(configTokenAuthMethod)) {
+    debug(
+      'Client authn method %o is not supported by the issuer. ' +
+        'Supported authn methods are %o.',
+      configTokenAuthMethod,
+      issuerTokenAuthMethods
+    );
+  }
+
+  if (config.clientAuthMethod == 'private_key_jwt') {
+    const configTokenAuthAlg = config.clientAssertionConfig.signingKey;
+    const issuerTokenAuthAlgs = Array.isArray(
+      issuer.token_endpoint_auth_signing_alg_values_supported
+    )
+      ? issuer.token_endpoint_auth_signing_alg_values_supported
+      : [];
+    if (!issuerTokenAuthAlgs.includes(configTokenAuthAlg)) {
+      debug(
+        'Client authn signing alg %o is not supported by the issuer. ' +
+          'Supported authn algs are %o.',
+        configTokenAuthAlg,
+        issuerTokenAuthAlgs
+      );
+    }
+  }
+
+  const jwks =
+    config.clientAuthMethod == 'private_key_jwt'
+      ? {
+          keys: [
+            JWK.asKey(config.clientAssertionConfig.signingKey, {
+              alg: config.clientAssertionConfig.signingAlg,
+              use: 'sig',
+            }).toJWK(true),
+          ],
+        }
+      : { keys: [] };
+  function getClientAuthMetdata(config) {
+    switch (config.clientAuthMethod) {
+      case 'none':
+        return {};
+      case 'private_key_jwt':
+        return {
+          token_endpoint_auth_signing_alg:
+            config.clientAssertionConfig.signingAlg,
+        };
+      case 'client_secret_basic':
+      case 'client_secret_post':
+        return { client_secret: config.clientSecret };
```

Comment:
Also , I think there's a use case for a client secret even when using `private_key_jwt` (when the ID Token signing alg is `HS256`) but I will confirm